Source/VCGeneration/Counterexample.cs

```diff
@@ -251,6 +240,22 @@ public void PrintModel(TextWriter tw, Counterexample counterexample)
       }
     }
 
+    public void InitializeStates()
+    {
+      if (Model is { ModelHasStatesAlready: false }) {
+        if (this is AssertCounterexample assertError) {
```

Comment:
Could we use a `case` instead of an if with `is` in each branch?

Reply:
Used an abstract method instead.